pkg/controllers/nodeclaim/termination/controller.go

```diff
@@ -90,7 +90,7 @@ func (c *Controller) Finalize(ctx context.Context, nodeClaim *v1beta1.NodeClaim)
 	}
 	if nodeClaim.Status.ProviderID != "" {
 		if err = c.cloudProvider.Delete(ctx, nodeClaim); cloudprovider.IgnoreNodeClaimNotFoundError(err) != nil || cloudprovider.IsRetryableError(err) {
-			return reconcile.Result{}, fmt.Errorf("terminating cloudprovider instance, %w", err)
+			return reconcile.Result{RequeueAfter: 10 * time.Second}, fmt.Errorf("terminating cloudprovider instance, %w", err)
```

Comment:
Can we special case this for the retryable error? For a transient error, I think that standard back-off is fine, but for this special type of error that we have, I'm thinking that's the only time that we should do this 10s back-off